syntax = "proto3";

package parser;

// 解析器服务定义
service ParserService {
  // 解析文件（主要接口）
  rpc ParseFile(ParseRequest) returns (ParseResponse);

  // 健康检查（Kubernetes 友好）
  rpc HealthCheck(HealthCheckRequest) returns (HealthCheckResponse);
}

// 解析请求消息
message ParseRequest {
  string file_path = 1;              // 文件路径
  string file_format = 2;            // 文件格式（pdf/docx/pptx/md）

  // 解析选项
  ParseOptions options = 3;
}

message ParseOptions {
  bool enable_ocr = 1;               // 是否启用 OCR（默认 true）
  bool enable_caption = 2;           // 是否启用 VLM Caption（默认 false）
  int32 max_image_size = 3;          // 最大图像尺寸（默认 4096px）
  string language = 4;               // OCR 语言（默认 "ch"）
}

// 解析响应消息
message ParseResponse {
  string content = 1;                // 解析后的文本内容
  ParseMetadata metadata = 2;        // 元数据
  string error_message = 3;          // 错误消息（如果失败）
}

message ParseMetadata {
  int32 page_count = 1;              // 页数（PDF/PPTX）
  int32 image_count = 2;             // 图像数量
  int32 table_count = 3;             // 表格数量
  int32 ocr_count = 4;               // OCR 识别的图像数量
  int32 caption_count = 5;           // Caption 生成的图像数量
  float parse_time_ms = 6;           // 解析耗时（毫秒）
}

// 健康检查请求
message HealthCheckRequest {
  string service = 1;
}

// 健康检查响应
message HealthCheckResponse {
  enum ServingStatus {
    UNKNOWN = 0;
    SERVING = 1;
    NOT_SERVING = 2;
    SERVICE_UNKNOWN = 3;
  }
  ServingStatus status = 1;
}
